modules/common/src/Util/DrupalFiles.php

```diff
@@ -101,19 +101,19 @@ public function getStreamWrapperManager(): StreamWrapperManager {
    * the newly stored file.
    */
   public function retrieveFile($url, $destination) {
```

Comment:
Can we just cast $url as string in the function declaration and not have to cast it everywhere it's used?

Reply:
You can't "cast" in a signature right? That's just validation I think? But we can add `$url = (string) $url;` at the top.

Reply:
You can say `public function retrieveFile(string $url, $destination) {` and that means `$url` is already a string inside the method, and then you won't have to cast it as a string to please the linter.

Reply:
Right; what I was concerned about was that `$url` could be another type that's cast-able to string, which would work as it is now but would throw an error if we added a type hint. But I was mostly thinking about `Drupal\Core\Url`, which a) I think is not actually cast-able to `string` (it has a `::toString()` but not a `::__toString()`), and b) the only place in DKAN this method is called is in `ResourceImporter`, where `$url` can only be a `string`. So I think it's ok to type hint here.